include/prometheus_gauge_populator.h

```diff
@@ -0,0 +1,32 @@
+#ifndef INCLUDE_PROMETHEUS_GAUGE_POPULATOR_H
+#define INCLUDE_PROMETHEUS_GAUGE_POPULATOR_H
+
+#include <gnuradio/block.h>
+#include <gnuradio/blocks/api.h>
+
+#include <prometheus/gauge.h>
+
+namespace gr {
+namespace prometheus {
+
+/// This block takes a float as an input and writes in into a prometheus gauge
+class prometheus_gauge_populator : virtual public block {
+public:
+  using sptr = boost::shared_ptr<prometheus_gauge_populator>;
```

Comment:
no just no lets use `std::share_ptr` or is there an actually convincing reason? 

or are we forced here to use a boost smart pointer to get the datafrom gnuradio?

Reply:
we need to use what gnuradio does, sadly